### PR TITLE
feat: share the challenge and fixed typo issue

### PR DESCRIPTION
**Pull request by Victorportus**
Share button on the challenge. 

if you are running the BE on localhost, the address will be undefined followed by the info. if running on the production server it will be something similar to this

https://weaup-mobile.web.app/challenge.html?id=669084bd278b084c69f8e430&user=667fc840b19a491b6f7ef6cf

### CODE REVIEW

**Comment by zeckkimon** [comment]
Task linked: [CU-86b15v2ym Add share action to challenge details page](https://app.clickup.com/t/86b15v2ym)

File: src/app/challenges/challenge-details.tsx [hunk 1]
```diff
@@ -37,6 +54,20 @@ const ChallengeDetail = () => {
...
+          {isOngoing && (
+            <Stack
+              w={40}
+              h={40}
+              backgroundColor={theme.colors.white}
+              alignItems="center"
+              justifyContent="center"
+              borderRadius={40}
+            >
+              <Pressable onPress={handleShare}>
```
Comment (whamester) [inline src/app/challenges/challenge-details.tsx]: can you use TouchableOpacity instead of Pressable?

**Review by whamester — APPROVED** [review]

File: src/app/challenges/challenge-details.tsx [hunk 2]
```diff
@@ -28,6 +31,20 @@ const ChallengeDetail = () => {
   const color = data?.data.color ?? theme.colors.secondary[100];
   const { isOngoing } = isChallengeActive(data?.data?.end_at);
 
+  const handleShare = async () => {
+    if (data) {
+      const urlWithUserId = data.data.url.replace("[user_id]", loggedUser);
+      try {
+        await Share.share({
+          message: urlWithUserId, // Android
+          url: urlWithUserId, // iOS
```
Comment (ReinhardtBotha) [inline src/app/challenges/challenge-details.tsx]: comment out this line, otherwise it sends the url x 2 on ios

**Review by ReinhardtBotha — APPROVED** [review]

**Review by zeckkimon — APPROVED** [review]

File: src/app/challenges/challenge-details.tsx (hunk 1 above)
Comment (Victorportus) [reply src/app/challenges/challenge-details.tsx]: changed as requested

File: src/app/challenges/challenge-details.tsx (hunk 2 above)
Comment (Victorportus) [reply src/app/challenges/challenge-details.tsx]: Fixed according to platform

**Pull request merged**